### ISSUE TITLE
Enabling monitoring didn't install the plugin on the nodes

### ISSUE THREAD

**Issue by RandellP** [Opster/opensearch-k8s-operator#586]
I am using the 2.3.1 version of the operator after upgrading from 2.2.1 (had to do 2.3.0 in the middle to get it to work right)

I followed the instructions at the bottom of the readme to setup the monitoring for the cluster.
It says

"If enabled the operator will install the aiven plugin into the opensearch pods and generate a Prometheus ServiceMonitor object to configure the plugin for scraping."

The service monitor is there.  The config on the opensearch operator has the monitoring section as expected and all that.
But the plugin is not on the nodes.  (not in the plugins dir on the nodes, and the nodes say there is no handler for /_prometheus/metrics).

I tried to read through the PR that added this feature, and I didn't see anything that would seem like it would trigger an install, but I don't know much go, so I could easily have miss it.

Is it expected to update an existing cluster? Or only for a new nodes?

**Comment by swoehrl-mw** [comment]
Hi @RandellP. The monitoring feature should also work for existing clusters. My expectation is that the operator performs a rolling restart of all nodes to apply the new config and install the plugin. The trigger happens generically by the operator updating the StatefulSet and then detecting that the pods are not up-to-date.

Can you please check the statefulsets of the nodepool and verify the following?:
* is the prometheus plugin part of the `opensearch` container command?
* is `status.updatedReplicas` equal to the configured number of replicas?
* Is the Opensearch version the expected one?

Also can you please check the operator logs for any errors and the opensearchcluster custom object status if a rollingRestart is in progress?

**Comment by RandellP** [comment]
@swoehrl-mw   This is interesting.  The statefulset just runs an sh script that looks like it must be built into the image.  And that script sets up the security plugin and a performance analyzer.  I don't see anything that would dynamically pull in a plugin either.  
And more importantly, the SS age is 6days while I did the changes yesterday.  So I don't think it got the update.

```
opensearch:
    Image:       docker.io/opensearchproject/opensearch:2.5.0
    Ports:       9200/TCP, 9300/TCP
    Host Ports:  0/TCP, 0/TCP
    Command:
      /bin/bash
      -c
      ./opensearch-docker-entrypoint.sh
```

The status.updateReplicas does match the number of data nodes.
The opensearch version hasn't changed as part of this upgrade, and it is what is expected to be.

The operator logs show no issues and in fact report that "resource is in sync" for the stateful set.  The logs did find a diff on the deployment for the dashboard and reported it being updated and now being in sync. 

And the image for the operator is 2.3.1 so it got updated.

So looking at the stateful set, I am not sure what update the operator should be doing.  The sh script doesn't appear to take any arguments, and the image seems tied to the database version.

Is it that the db version of 2.5.0 has to be updated as well as the operator version to support the monitoring?

**Comment by swoehrl-mw** [comment]
Hi @RandellP. Found the problem. There was a bug in the initial implementation of the monitoring feature that lead to the plugin not being installed on the nodes. The fix is already on main but not yet in a release (issue #487, PR #490).

**Comment by RandellP** [comment]
Great, thanks for finding and fixing it.  I will do a manual install of the plugin and work form that until the release.

**Comment by RandellP** [comment]
@swoehrl-mw So I got sidetracked and the release came out before I manually installed the plugins.  So I updated the operator to the new release.  Then applied the change to enable monitoring.  This time it detected a diff in the deployment for the dashboard and updated that.  But nothing for the nodes.  And looking in the plugins dir on the nodes, I don't see anything that looks like monitoring.

**Comment by Alwinius** [comment]
This fix is part of release 2.4.0 in case someone was waiting for this like me.
